### PR TITLE
Add s3 archive example plugin

### PR DESCRIPTION
**Pull request by crenshaw-dev**

### CODE REVIEW

File: cmd/argo/commands/plugin/plugin.go [hunk 1]
```diff
@@ -64,10 +65,13 @@ func loadPlugin(pluginDir string) (*loadResult, error) {
...
-	runAsNonRoot := true
-	runAsUser := int64(1000)
-	plug.Spec.Container.SecurityContext = &apiv1.SecurityContext{RunAsNonRoot: &runAsNonRoot, RunAsUser: &runAsUser}
+	if strings.Contains(pluginDir, "controller") {
+		// Match default security settings for easier patch.
+		runAsNonRoot := true
+		runAsUser := int64(1000)
+		plug.Spec.Container.SecurityContext = &apiv1.SecurityContext{RunAsNonRoot: &runAsNonRoot, RunAsUser: &runAsUser}
+	}
+
```
Comment (simster7) [inline cmd/argo/commands/plugin/plugin.go:74]: Should this be its own PR? Seems separate from your example

Comment (crenshaw-dev) [reply cmd/argo/commands/plugin/plugin.go:74]: Fair, will do.

Comment (crenshaw-dev) [reply cmd/argo/commands/plugin/plugin.go]: https://github.com/argoproj/argo-workflows/pull/7254
